File: src/llist.py
class Element:
    def __init__(self, data):
        self.data = data
        self.link = None

class LList:    
    def __init__(self):
        self.root = Element(None)

    def _get_last(self):
        last = self.root
        while last.link is not None:
            last = last.link
        return last

    def prepend(self, data):
        elem = Element(data)
        elem.link = self.root.link
        self.root.link = elem

    def append(self, data):
        last = self._get_last()
        last.link = Element(data)
        
    def show(self):
        elem = self.root
        while True:
            elem = elem.link
            if elem is None:
                break
            print(elem.data)
            
llist = LList()
llist.append(123)
llist.prepend(456)
llist.append(789)
llist.show()

File: src/llist.rs
struct Element<'a> {
    data: i32,
    link: Option<&'a Box<Element<'a>>>,
}

impl Element<'_> {
    fn new(data: i32) -> Self {
        Self{data: data, link: None}
    }
}

struct LList<'a> {
    root: &'a mut Box<Element<'a>>,
}

impl LList<'_> {
    fn new() -> Self {
        Self{root: & Box::new(Element::new(0))}
    }
    
    fn prepend(&mut self, data: i32) {
        let mut elem = Element::new(data);
        elem.link = self.root.link;
        self.root.link = Some(&Box::new(elem));
    }
    /*
    fn append(data: i32) {
    }*/
    /*
    fn show(&self) {
        let mut elem = &self.root;
        loop {
            let maybe_elem = &elem.link;
             match maybe_elem {
                 None => { break; }
                 Some(some_elem) => {
                     elem = &some_elem;
                     println!("{}", elem.data)
                 }
             }
        }
    }*/
}

fn main() {
    let mut llist = LList::new();
    //llist.append(123);
    llist.prepend(456);
    llist.prepend(99999999);
    //llist.append(789);
    //llist.show();
}
